From 7dca17c21cb6a8ffd8fe9964812c8f0447c4ab7b Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Fri, 9 Oct 2015 05:57:51 +0200
Subject: [PATCH 1/2] makepkg * deprecate 'asneeded' and add --as-needed to be
 default... * WTF.. adding 8k lines with this option.. * use noasneeded if you
 need an package to be build without

---
 scripts/makepkg | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index d070fe0..8fff977 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -32,6 +32,7 @@ PKGDEST=$startdir
 USE_COLOR="n"
 
 # Options
+
 BUILDSCRIPT="./FrugalBuild"
 CLEANUP=0
 CLEANCACHE=0
@@ -1342,16 +1343,23 @@ if grep -i -q "^# Compiling Time: [~0-9\.]\+ SBU$" $BUILDSCRIPT && \
 	fi
 fi
 
+
+if [ "`check_option ASNEEDED`" ]; then
+        error "You are using deprecated option 'asneeded'."
+        error "This is now default, please remove and retry."
+	exit 1
+fi
+
 if [ "`check_option NOASNEEDED`" ]; then
-	msg "Using --no-as-needed in LDFLAGS"
+	warning "Using --no-as-needed in LDFLAGS ( are you sure ? )"
 	LDFLAGS+=" -Wl,--no-as-needed"
+	export LDFLAGS
 else
-	if [ "`check_option ASNEEDED`" ]; then
-		msg "Using --as-needed in LDFLAGS"
-		LDFLAGS+=" -Wl,--as-needed"
-	fi
+	msg "Using --as-needed in LDFLAGS ( default )"
+	LDFLAGS+=" -Wl,--as-needed"
+	export LDFLAGS
 fi
-export LDFLAGS
+
 
 # retrieve sources
 msg "Retrieving Sources..."
-- 
2.5.0

